### PR TITLE
Macro feature

### PR DESCRIPTION
**Pull request by pawel-wozniak**

### CODE REVIEW

File: src/main/java/edu/kis/powp/jobs2d/features/RecordingFeature.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+				.flatMap(x -> Arrays.stream(x.getSubElements()))
+				.flatMap(x -> Arrays.stream(x.getSubElements()))
+				.filter(x -> x instanceof JCheckBoxMenuItem)
+				.map(x -> (JCheckBoxMenuItem) x)
+				.filter(x -> x.getActionCommand().equals(RECORDING))
+				.findFirst()
+				.orElseGet(JCheckBoxMenuItem::new);
+	}
+
+	public static void register() {
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/features/RecordingFeature.java]: Why is it public? 
The name is a little bit confusing.

File: src/main/java/edu/kis/powp/jobs2d/drivers/decorator/RecordingDriverDecorator.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+	@Override
+	public void operateTo(int x, int y) {
+		list.add(new OperateToCommand(x, y));
+		driver.operateTo(x,y);
+	}
+
+	@Override
+	public void update() {
+		Job2dDriver newDriver = DriverFeature.getDriverManager().getCurrentDriver();
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/decorator/RecordingDriverDecorator.java]: Unnecessary dependency with DriverFeature.

File: src/main/java/edu/kis/powp/jobs2d/drivers/decorator/RecordingDriverDecorator.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+import edu.kis.powp.jobs2d.features.DriverFeature;
+import edu.kis.powp.jobs2d.features.RecordingFeature;
+import edu.kis.powp.observer.Subscriber;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+
+public class RecordingDriverDecorator implements Job2dDriver, Subscriber {
```
Comment (kowallus) [inline src/main/java/edu/kis/powp/jobs2d/drivers/decorator/RecordingDriverDecorator.java]: SRP rule. Could you separate Subscriber from Job2dDriver?

**Review by kowallus — COMMENTED** [review]
Please do some code cleaning.

File: src/main/java/edu/kis/powp/jobs2d/features/RecordingFeature.java (hunk 1 above)
Comment (pawel-wozniak) [reply src/main/java/edu/kis/powp/jobs2d/features/RecordingFeature.java]: Changed to private and renamed

File: src/main/java/edu/kis/powp/jobs2d/drivers/decorator/RecordingDriverDecorator.java (hunk 2 above)
Comment (pawel-wozniak) [reply src/main/java/edu/kis/powp/jobs2d/drivers/decorator/RecordingDriverDecorator.java]: Dependency removed

File: src/main/java/edu/kis/powp/jobs2d/drivers/decorator/RecordingDriverDecorator.java (hunk 3 above)
Comment (pawel-wozniak) [reply src/main/java/edu/kis/powp/jobs2d/drivers/decorator/RecordingDriverDecorator.java]: Moved Subscriber logic to separate class

**Pull request merged**